libr/core/cio.c

```diff
@@ -367,18 +367,45 @@ R_API int r_core_seek_delta(RCore *core, st64 addr) {
 	return ret;
 }
 
+static char * _patch_comment(RCore *core, ut64 addr, int size) {
+	char *out = NULL;
+
+	ut8 *bytes = malloc (size);
+	int cnt = r_io_read_at (core->io, addr, bytes, size);
+	char *hex_pairs = r_hex_bin2strdup (bytes, cnt);
+	free (bytes);
+
+	out = malloc (snprintf (NULL, 0, "patch: %d bytes (%s)", cnt, hex_pairs) + 1);
```

Comment:
You talk about `snprintf` that can return negative value? Or you talk about missed null checking?